src/util_types/mutator_set/removal_record.rs

```diff
@@ -254,20 +255,25 @@ impl RemovalRecord {
         let mut mmr_membership_proofs_for_append: Vec<
             &mut mmr::mmr_membership_proof::MmrMembershipProof<Hash>,
         > = vec![];
+        let mut leaf_indices = vec![];
         for (i, rr) in removal_records.iter_mut().enumerate() {
             if rrs_for_batch_append.contains(&i) {
-                for (_, (mmr_mp, _chnk)) in rr.target_chunks.iter_mut() {
-                    mmr_membership_proofs_for_append.push(mmr_mp);
+                for (chunk_index, (mmr_mp, _chnk)) in rr.target_chunks.iter_mut() {
+                    if *chunk_index != old_window_start_batch_index {
```

Comment:
Again: Not sure about the need for this check. Covered by line 227: `if chunk_index < old_window_start_batch_index`?

Reply:
Same story: if I remove the condition then tests fail. But I'm puzzled as to why the line you point to doesn't catch it.